instrumentation/apache-dubbo-2.7/testing/src/main/java/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTest.java

```diff
@@ -0,0 +1,290 @@
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.opentelemetry.semconv.ServerAttributes;
+import io.opentelemetry.semconv.incubating.RpcIncubatingAttributes;
+import java.lang.reflect.Field;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.apache.dubbo.common.utils.NetUtils;
+import org.apache.dubbo.config.ApplicationConfig;
+import org.apache.dubbo.config.ProtocolConfig;
+import org.apache.dubbo.config.ReferenceConfig;
+import org.apache.dubbo.config.RegistryConfig;
+import org.apache.dubbo.config.ServiceConfig;
+import org.apache.dubbo.config.bootstrap.DubboBootstrap;
+import org.apache.dubbo.rpc.service.GenericService;
+import org.assertj.core.api.AbstractAssert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+abstract class AbstractDubboTest {
+
+  private final ProtocolConfig protocolConfig = new ProtocolConfig();
+
+  abstract InstrumentationExtension testing();
+
+  private DubboBootstrap bootstrap;
+  private DubboBootstrap consumerBootstrap;
+
+  @BeforeAll
+  static void setUp() throws Exception {
+    System.setProperty("dubbo.application.qos-enable", "false");
+    Field field = NetUtils.class.getDeclaredField("LOCAL_ADDRESS");
+    field.setAccessible(true);
+    field.set(null, InetAddress.getLoopbackAddress());
+  }
+
+  @AfterAll
+  static void tearDown() {
+    System.clearProperty("dubbo.application.qos-enable");
+  }
+
+  @AfterEach
+  @SuppressWarnings("CatchingUnchecked")
+  public void afterEach() {
+    List<DubboBootstrap> dubboBootstraps = Arrays.asList(bootstrap, consumerBootstrap);
+    for (DubboBootstrap bootstrap : dubboBootstraps) {
+      try {
+        if (bootstrap != null) {
+          bootstrap.destroy();
+        }
+      } catch (Exception e) {
+        // ignore
+      }
+    }
+  }
+
+  ReferenceConfig<HelloService> configureClient(int port) {
+    ReferenceConfig<HelloService> reference = new ReferenceConfig<>();
+    reference.setInterface(HelloService.class);
+    reference.setGeneric("true");
+    reference.setUrl("dubbo://localhost:" + port + "/?timeout=30000");
+    return reference;
+  }
+
+  ServiceConfig<HelloServiceImpl> configureServer() {
+    RegistryConfig registerConfig = new RegistryConfig();
+    registerConfig.setAddress("N/A");
+    ServiceConfig<HelloServiceImpl> service = new ServiceConfig<>();
+    service.setInterface(HelloService.class);
+    service.setRef(new HelloServiceImpl());
+    service.setRegistry(registerConfig);
+    return service;
+  }
+
+  @Test
+  @SuppressWarnings({"rawtypes", "unchecked"})
```

Comment:
as far as I can tell the only piece that does not type nicely is the conversion from `ReferenceConfig<HelloService>` to `ReferenceConfig< GenericService>`. I'd do that conversion in a helper method and add the warning suppression there.